File: solutions/1603-running-sum-of-1d-array/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* runningSum(int* nums, int size, int* returnSize) {
    int *arr = (int *)malloc(size*sizeof(int)); 
    int i; 
    arr[0] = nums[0]; 
    *returnSize = size; 
    for(i = 1; i < size; i++){
        arr[i] = nums[i] + arr[i-1]; 
    }
    return arr; 
}

File: solutions/1603-running-sum-of-1d-array/solution.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        ans = []
        son = nums[0]
        for x in range(len(nums)-1):
            ans.append(son)
            son += nums[x+1]
        ans.append(son)
        return ans
